src/world/World.cpp

```diff
@@ -243,6 +243,23 @@ bool World::IsPlayerTerritory(const MapPoint pt) const
     return true;
 }
 
+bool World::IsPlayerTerritory(const MapPoint pt, const unsigned char owner) const
```

Comment:
I would combine this with the function above. E.g `owner == 0` -> any owner. So just change the `if(ptOwner...` and call this function from the one above or use a default param. Don't forget the docu